spec/spec_helper.rb

```diff
@@ -34,16 +34,14 @@
 ActiveRecord::Migration.maintain_test_schema!
 
 RSpec.configure do |config|
+  config.expose_dsl_globally = false # disable RSpec < 3 global monkey patching
   config.use_transactional_fixtures = false
   config.infer_base_class_for_anonymous_controllers = false
 
   config.order = 'random'
 
   config.include ActiveJob::TestHelper
   config.include RSpecHtmlMatchers
-  config.extend ControllerMacros, type: :controller
-
-  config.infer_spec_type_from_file_location!
```

Comment:
I kinda like this auto-detection system 😢

Reply:
😿 
Interesting / promissing thing about the current setup: it *almost* works with transactional fixtures 🎉 
There have apparently been some changes to Rails' test internals that may allow to ditch `database_cleaner` and switch on `transactional_fixtures` instead. I tried it out - it runs super fast. But unfortunately some tests around the `season` model do not really work with this setup 😉